File: exercises/python_playwright/Hidden_Layers/hidden_layers_playwright.py
from playwright.sync_api import Page


class HiddenLayers:
    def __init__(self, page: Page):
        """Store the Playwright page and element locators."""
        self.page = page
        self.green_button = page.locator("#greenButton")


    def open(self) -> None:
        """Open the Hidden Layers exercise page."""
        self.page.goto("http://uitestingplayground.com/hiddenlayers")


    def click_green_button(self) -> None:
        """Click the green button."""
        self.green_button.click(timeout=3000)

File: exercises/javascript_playwright/hidden-layers/hiddenLayersPlaywright.js
export class HiddenLayersPage {
    /**
     *
     * @param {import('playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.greenButton = page.locator("#greenButton");
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/hiddenlayers");
    }

    async clickGreenButton() {
        await this.greenButton.click();
    }

    async isGreenButtonCovered() {
        return await this.greenButton.evaluate((button) => {
            const buttonRect = button.getBoundingClientRect();
            const centerX = buttonRect.left + buttonRect.width / 2;
            const centerY = buttonRect.top + buttonRect.height / 2;

            const topElement = document.elementFromPoint(centerX, centerY);
            return topElement !== button;
        });
    }
}
